src/index.ts

```diff
@@ -0,0 +1,48 @@
+import * as mongoose from 'mongoose';
+import { HealthCheckResponse } from 'grpc-ts-health-check';
+import { Server, serviceNames } from './server';
+import { config } from './config';
+import { log, Severity } from './utils/logger';
+import { getCurrTraceId } from './utils/wrapper';
+
+
+(async () => {
+    const server: Server = new Server(config.server.bindAddress);
+    await connectMongo(server);
+    server.server.start();
+})();
+
+async function connectMongo(server: Server) {
```

Comment:
Maybe that should also be part of the `Server` class.

Reply:
Or at least a class that extends `Server`

Reply:
I don't think that mongo connection is related to the server. 
The only reason I pass server as param is that i need to set health status when the connection to mongo fails.